### PR TITLE
Avoid API request for API.getPagesComparisonsDisabledFor on Login pages

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

While testing another issue I saw that on login pages a request to `API.getPagesComparisonsDisabledFor` was sent. This is actually unneeded.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/CoreHome/vue/src/Comparisons/Comparisons.store.ts [hunk 1]
```diff
@@ -77,7 +77,9 @@ export default class ComparisonsStore {
   readonly isEnabled = computed(() => this.checkEnabledForCurrentPage());
 
   constructor() {
-    this.loadComparisonsDisabledFor();
+    document.addEventListener('DOMContentLoaded', () => {
+      this.loadComparisonsDisabledFor();
+    });
```
Comment (sgiehl) [inline plugins/CoreHome/vue/src/Comparisons/Comparisons.store.ts]: This change was needed as `window.broadcast.isLoginPage()` checks if the body element has a certain id. But the javascript might currently already be parsed before the browser starts to render the html, which makes it impossible to look for html elements id.
Not sure if there is a smarter approach for this...

**Comment by diosmosis** [comment]
haven't tested it, but the code looks fine

File: plugins/CoreHome/vue/src/Comparisons/Comparisons.store.ts (hunk 1 above)
Comment (diosmosis) [reply plugins/CoreHome/vue/src/Comparisons/Comparisons.store.ts]: just a thought that came up, but an alternative might be to make `Piwik::getLoginPluginName()` a global in jsGlobalVariables.twig and check if the URL module == that or `'Login'`

**Review by bx80 — APPROVED** [review]
This worked well for me when testing, the unnecessary API call is not made when loading the login page :+1:

* [x] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [x] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [x] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [x] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [x] n/a [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [x] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [x] n/a [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [x] none [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [x] n/a [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [x] n/a [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [x] n/a Existing documentation updated if needed

**Pull request merged**